### PR TITLE
Information_Gain Function to return Best feature and Threshhold

### PR DESCRIPTION
**Pull request by arkadeep16121993**

### CODE REVIEW

**Comment by marleenlki** [comment]
@arkadeep16121993 , I had similar issue with the package dependencies. Try switching to your test directory and activate the test environment with Pkg.activate("."). Then you add the missing package to the test environment Pkg.add("Random"). At least for me it worked afterwards.

File: Information_Gain_Arka.jl [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+using Random
+using Statistics
+using StatsBase  # Importing StatsBase for the countmap function
+
+"""
+    entropy(y::Vector)
+
+Calculate the entropy of a vector of labels `y`.
+"""
+function entropy(y::Vector)
```
Comment (Enea09) [inline Information_Gain_Arka.jl]: use parametric types. For example: function entropy(y::Vector{T}) where T

Comment (Enea09) [reply Information_Gain_Arka.jl]: specify the parameters in the docstrings for clarity: y: A vector of labels

File: Information_Gain_Arka.jl [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    return -sum(p -> p > 0 ? p * log2(p) : 0, probs)
+end
+
+"""
+    information_gain(y::Vector, y_left::Vector, y_right::Vector)
+
+Calculate the Information Gain of a split. `y` is the original labels vector, `y_left` and `y_right` are the labels vectors
+for the left and right splits, respectively.
+"""
+function information_gain(y::Vector, y_left::Vector, y_right::Vector)
```
Comment (Enea09) [inline Information_Gain_Arka.jl]: function information_gain(y::Vector{T}, y_left::Vector{T}, y_right::Vector{T}) where T

File: Information_Gain_Arka.jl [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    counts = countmap(y)
+    probs = values(counts) ./ length(y)
+    return -sum(p -> p > 0 ? p * log2(p) : 0, probs)
+end
+
+"""
+    information_gain(y::Vector, y_left::Vector, y_right::Vector)
+
+Calculate the Information Gain of a split. `y` is the original labels vector, `y_left` and `y_right` are the labels vectors
+for the left and right splits, respectively.
```
Comment (Enea09) [inline Information_Gain_Arka.jl]: specify what the function return as well in the docstring

File: Information_Gain_Arka.jl [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    return H - (p_left * H_left + p_right * H_right)
+end
+
+"""
+    split_dataset(X::Matrix, y::Vector, feature::Int, threshold::Real)
+
+Split the dataset `X` and labels `y` based on a `feature` and a `threshold`.
+Returns the left and right splits for both `X` and `y`.
+"""
+function split_dataset(X::Matrix, y::Vector, feature::Int, threshold::Real)
```
Comment (Enea09) [inline Information_Gain_Arka.jl]: does this handle non-numerical features?

File: Information_Gain_Arka.jl [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    return X_left, y_left, X_right, y_right
+end
+
+"""
+    best_split(X::Matrix, y::Vector)
+
+Find the best split for the dataset `X` and labels `y` based on Information Gain.
+Returns the best feature and threshold for the split.
+"""
+function best_split(X::Matrix, y::Vector)
```
Comment (Enea09) [inline Information_Gain_Arka.jl]: function best_split(X::Matrix{T}, y::Vector{T}) where T

File: Information_Gain_Arka.jl (hunk 4 above)
Comment (arkadeep16121993) [reply Information_Gain_Arka.jl]: Yes. You can check the test file

**Comment by Enea09** [comment]
Some general comments: 
place the Information_Gain_Arka file in the src directory
follow Julia's naming conventions: for example rename the module to InformationGain
include the package into the DecisionTreeAndRandomForest module: include(“Information_Gain_Arka.jl")

File: test/Information_Gain_test.jl [hunk 6]
```diff
@@ -0,0 +1,53 @@
+using Random
+using Statistics
+using StatsBase  # Importing StatsBase for the countmap function..
+using Test
+
+# Import the functions from your Information Gain implementation..
+include("..\\Information_Gain_Arka.jl")
```
Comment (Enea09) [inline test/Information_Gain_test.jl:7]: you can add this to run_tests.jl: 
@testset "DecisionTreeAndRandomForest.jl" begin
    include("Information_Gain_test.jl")
end

**Pull request closed without merging**